modules/affile/logproto-file-writer.h

```diff
@@ -25,6 +25,15 @@
 #define LOG_PROTO_FILE_WRITER_H_INCLUDED
 
 #include "logproto/logproto-client.h"
+#include "logthrdest/logthrdestdrv.h"
+#include "modules/file-rotation/file-rotation.h"
+
+typedef struct _FileRotationRequestData FileRotationRequestData;
```

Comment:
Let's call this `FileFlushSignalData` for now.

The signal needs to be declared on this side, not in the file rotation plugin.
For that reason, let's call it `signal_file_flush`.

(We try to avoid mentioning file rotation here, we just provide generic signals, so any plugin can attach to them and do their job.)

All of these declarations should go into a separate header file with an LGPL license header.
We could call it `file-signals.h`.
An example can be found here: https://github.com/syslog-ng/syslog-ng/blob/master/modules/http/http-signals.h

Reply:
Created :+1: